tests/nodeunit/unit/validate_doc_update.js

```diff
@@ -29,22 +19,64 @@ exports.setUp = function(cb) {
   cb();
 };
 
+var testAllowed = function(roles, newDoc, oldDoc) {
+  oldDoc = oldDoc || newDoc;
+  var userCtx = { name: 'a-user', roles: roles };
+  return lib._allowed(newDoc, oldDoc, userCtx, {}).allowed;
+};
+
+exports['only db and national admins are allowed change ddocs'] = function(test) {
+  var doc = { _id: '_design/something' };
+  test.equal(testAllowed([ '_admin' ], doc), true);
+  test.equal(testAllowed([ 'national_admin' ], doc), true);
+  test.equal(testAllowed([ ], doc), false);
+  test.done();
+};
+
+exports['only db and national admins are allowed change forms'] = function(test) {
+  var doc = { type: 'form' };
+  test.equal(testAllowed([ '_admin' ], doc), true);
+  test.equal(testAllowed([ 'national_admin' ], doc), true);
+  test.equal(testAllowed([ ], doc), false);
+  test.done();
+};
+
+exports['only db and national admins are allowed change translations'] = function(test) {
+  var doc = { type: 'form' };
+  test.equal(testAllowed([ '_admin' ], doc), true);
+  test.equal(testAllowed([ 'national_admin' ], doc), true);
+  test.equal(testAllowed([ ], doc), false);
+  test.done();
+};
+
+exports['only db and national admins are allowed change resources'] = function(test) {
+  var doc = { type: 'form' };
+  test.equal(testAllowed([ '_admin' ], doc), true);
+  test.equal(testAllowed([ 'national_admin' ], doc), true);
+  test.equal(testAllowed([ ], doc), false);
+  test.done();
+};
+
+exports['only db and national admins are allowed change appcache doc'] = function(test) {
+  var doc = { type: 'form' };
```

Comment:
ahem.

Sir. Once one copies and once one pastes, one must also edit for relevance!